one-ui/src/main/java/com/marklogic/hub/oneui/controllers/EnvironmentController.java

```diff
@@ -112,8 +115,13 @@ public void onError(String commandName, Exception exception) {
         String directory = payload.get("directory").asText("");
         // setting the project directory will resolve any relative paths
         try {
+            Path directoryPath = Paths.get(directory);
             if (StringUtils.isEmpty(directory)) {
                 throw new BadRequestException("Missing required directory property");
+            } else if (!directoryPath.isAbsolute()) {
+                throw new ProjectDirectoryException("Directory supplied must be an absolute path");
+            } else if (!directoryPath.toFile().exists()) {
+                throw new ProjectDirectoryException("Directory '" + directory + "' does not exist or cannot be read");
```

Comment:
I'd clarify this too as "Project directory".

Reply:
@ryanjdew before this check we could provide a path and the missing directory would be created during install. There is no requirement around that. But this implementation now enforces a directory to exist. Is that something we want to do?

Also @wooldridge, these exceptions are thrown as a "message" and UI shows "suggestion" for all the above conditions -- "Verify that the user account running the service has permissions to read from or write to the project directory."

This install-form.tsx snippet might have to change if we want the message to show up in GUI:

```
let message = (error.response.status === 500) ? 'Internal Server Error' : error.response.data.message;
      setIsLoading(false);
      setErrorMessage({
        show: true, 
        message: 'Installation Failure', 
          description: <><p>{message}</p><p>{error.response.data.suggestion}</p></>
      });
```